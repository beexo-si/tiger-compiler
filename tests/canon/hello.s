.data
msg:
	.ascii    "Hello, world!\n"
	len = . - msg
.text
.global main
main:
	push %ebp
	mov %esp, %ebp
	push $msg
	call printf
	add $4, %esp
	mov %ebp, %esp
	pop %ebp
	ret
