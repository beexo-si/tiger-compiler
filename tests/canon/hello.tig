/*TIGER_ARGS=-arbol -ir -canon*/
(print ("Hello World");0)
