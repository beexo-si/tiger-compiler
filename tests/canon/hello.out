SeqExp([CallExp{func=print,args=["Hello World"]}0])
bien!
L1:	.quad 11
	.string "Hello World"
;;-PROC-L0__tigermain_0--:
SEQ(
 MOVE(
  TEMP T2,
  TEMP rbx),
 SEQ(
  MOVE(
   TEMP T3,
   TEMP r12),
  SEQ(
   MOVE(
    TEMP T4,
    TEMP r13),
   SEQ(
    MOVE(
     TEMP T5,
     TEMP r14),
    SEQ(
     MOVE(
      TEMP T6,
      TEMP r15),
     SEQ(
      MOVE(
       MEM(
        BINOP(PLUS,
         TEMP rbp,
         CONST -8)),
       TEMP rdi),
      SEQ(
       MOVE(
        TEMP rax,
        ESEQ(
         SEQ(
          MOVE(
           TEMP T0,
           NAME L1),
          EXP(
           CALL(
            NAME print,
             TEMP T0))),
         CONST 0)),
       SEQ(
        MOVE(
         TEMP rbx,
         TEMP T2),
        SEQ(
         MOVE(
          TEMP r12,
          TEMP T3),
         SEQ(
          MOVE(
           TEMP r13,
           TEMP T4),
          SEQ(
           MOVE(
            TEMP r14,
            TEMP T5),
           MOVE(
            TEMP r15,
            TEMP T6))))))))))))
;;-END-PROC--:
------Canon------
L1:	.quad 11
	.string "Hello World"
;;--END-STRS--:
--FRAME L0__tigermain_0:
LABEL L3
MOVE(
 TEMP T2,
 TEMP rbx)
MOVE(
 TEMP T3,
 TEMP r12)
MOVE(
 TEMP T4,
 TEMP r13)
MOVE(
 TEMP T5,
 TEMP r14)
MOVE(
 TEMP T6,
 TEMP r15)
MOVE(
 MEM(
  BINOP(PLUS,
   TEMP rbp,
   CONST -8)),
 TEMP rdi)
MOVE(
 TEMP T0,
 NAME L1)
EXP(
 CALL(
  NAME print,
   TEMP T0))
MOVE(
 TEMP rax,
 CONST 0)
MOVE(
 TEMP rbx,
 TEMP T2)
MOVE(
 TEMP r12,
 TEMP T3)
MOVE(
 TEMP r13,
 TEMP T4)
MOVE(
 TEMP r14,
 TEMP T5)
MOVE(
 TEMP r15,
 TEMP T6)
JUMP( NAME L2)
LABEL L2
;;-END-FRAME-:
yes!!
